### ISSUE TITLE
Generated .beads/.gitignore omits .br_recovery/, leaving recovery artifacts untracked-visible

### ISSUE THREAD

**Issue by rnybergh** [Dicklesworthstone/beads_rust#271]
## Problem

`br init` writes `.beads/.gitignore` with an entry for `.br_history/` but no entry for `.br_recovery/`. After any event that writes recovery artifacts (DB family quarantine, truncated WAL preservation, etc.), `git status` fills up with dozens of untracked files like:

```
.beads/.br_recovery/beads.db-shm.20260501_120551_758437000.truncated-wal
.beads/.br_recovery/beads.db-wal.20260501_120551_758437000.truncated-wal
...
```

The generic `*.db-shm` / `*.db-wal` patterns earlier in the file don't catch them because the recovery artifacts have a `.truncated-wal` suffix appended after the timestamp, so they don't match `*.db-wal`.

In one workspace I ended up with ~60 of these visible to git after a normal day of work.

## Repro

1. `br init` in a fresh repo
2. Trigger any condition that produces recovery artifacts (or just create a file at `.beads/.br_recovery/foo.truncated-wal` to demonstrate)
3. `git status` shows it as untracked

## Source

The template lives in [`src/cli/commands/init.rs`](https://github.com/Dicklesworthstone/beads_rust/blob/main/src/cli/commands/init.rs#L125-L126):

```
# Local history backups
.br_history/
```

Per `docs/ARCHITECTURE.md` and `docs/TROUBLESHOOTING.md`, `.br_recovery/` is documented as a sibling local-only directory for quarantining the DB family — same lifecycle as `.br_history/`, so it should be in the same template.

## Suggested fix

Add `.br_recovery/` next to `.br_history/` in the init template:

```
# Local history backups
.br_history/
.br_recovery/
```

Existing workspaces would also need to add it manually, since `br init` only writes the file when it doesn't already exist (the `gitignore_existed` branch). Worth considering whether `br doctor` should detect-and-offer to append missing entries.

Happy to send a PR if useful.

**Comment by Dicklesworthstone** [comment]
Fixed in [\`83239b6c\`](https://github.com/Dicklesworthstone/beads_rust/commit/83239b6c). Added \`.br_recovery/\` to the init template at \`src/cli/commands/init.rs:128-132\` next to the existing \`.br_history/\` entry, with a comment that calls out the \`.truncated-wal\` suffix specifically so future maintainers don't strip it under a "covered by `*.db-wal`" assumption. Golden snapshot at \`tests/snapshots/golden_beads_init__init_text_contents.snap\` updated in lockstep so \`cargo insta\` stays green.

Existing workspaces aren't modified by `br init` (it only writes when the file is absent), so any repo that already has a \`.beads/.gitignore\` needs the line appended manually:

```bash
printf '\n# DB-family recovery artifacts (truncated WAL/SHM, quarantined sidecars)\n.br_recovery/\n' >> .beads/.gitignore
```

You raised whether \`br doctor\` should detect-and-offer the append. That's a clean follow-up — the doctor already has the precedent (the gitignore-fix branch in `main.rs:3085`-`3093` calls \`fix_root_gitignore_if_warned\` for the parent repo's gitignore). Worth filing as a separate issue if you want it; this one is closed on the init template fix.
